packages/pigeon/lib/dart_generator.dart

```diff
@@ -8,10 +8,13 @@ import 'generator_tools.dart';
 /// Options that control how Dart code will be generated.
 class DartOptions {
   /// Constructor for DartOptions.
-  DartOptions({this.isNullSafe = true});
+  DartOptions({this.isNullSafe = true, this.copyrightHeader});
 
   /// Determines if the generated code has null safety annotations (Dart >=2.12 required).
   bool isNullSafe;
+
+  /// Path to a copyright header that will get prepended to generated code.
+  Iterable<String>? copyrightHeader;
```

Comment:
not clear from the command line arguments why it's an iterable

Reply:
Updated the comment, fwiw `Iterable<String>` is an interface that works over files or memory which makes it easier to test.  It could be a String, but that forces you to be memory bound.  Although thats probably not going to be an issue here.